components/RefinementList.js

```diff
@@ -19,7 +19,10 @@ class RefinementList extends React.Component {
   // has a checkbox inside, we ignore the first click event because we will get another one.
   handleClick(value, e) {
     if (e.target.tagName === 'A' && e.target.href) {
+      // do not trigger any url change by the href
       e.preventDefault();
+      // do not bubble (so that hierarchical lists are not triggering refine twice)
+      e.stopPropagation();
```

Comment:
Would it be possible to redispatch the event from the root refinement list in order to let the user listen to it? (just an idea)